src/events.js

```diff
@@ -3192,172 +3150,116 @@ export const populateTrackingQuery = async (boxIdAndBagsObj) => {
-            let shipmentCourier = document.getElementById('courierSelect').value;
-            finalShipmentTracking(boxIdAndBagsObj, userName, boxWithTempMonitor, shipmentCourier);
-        }
+        localforage.setItem("shipData", shippingData);
+        const shipmentCourier = document.getElementById('courierSelect').value;
+        finalShipmentTracking({boxIdAndBagsObj, boxIdAndTrackingObj, userName, boxWithTempMonitor, shipmentCourier});
     })
 
 }
 
 export const addEventSaveButton = async (boxIdAndBagsObj) => {
     document.getElementById('saveTracking').addEventListener('click', async () => {
-        let boxIdArray = Object.keys(boxIdAndBagsObj).sort(compareBoxIds);
-        let isMismatch = -1;
+        let isMismatch = false;
+        let shippingData = [];
+        let boxIdAndTrackingObj = {};
+        const boxIdArray = Object.keys(boxIdAndBagsObj);
 
-        for (let i = 0; i < boxIdArray.length; i++) {
-            let boxi = document.getElementById(boxIdArray[i] + "trackingId").value.toUpperCase();
-            let boxiConfirm = document.getElementById(boxIdArray[i] + "trackingIdConfirm").value.toUpperCase();
-            
-            if (boxi !== boxiConfirm) {
-                isMismatch = i;
-                break;
+        for (const boxId of boxIdArray) {
+            const trackingId = document.getElementById(boxId + "trackingId").value.toUpperCase();
+            const trackingIdConfirm = document.getElementById(boxId + "trackingIdConfirm").value.toUpperCase();
+    
+            if (trackingId !== trackingIdConfirm) {
+              isMismatch = true;
+              break;
             }
 
-            // if '959708259' exists update tracking number
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('959708259')) {
-              boxIdAndBagsObj[boxIdArray[i]]['959708259'] = boxi
-            }
-            // if 'confirmTrackNum' exists update tracking number
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('confirmTrackNum')) {
-              boxIdAndBagsObj[boxIdArray[i]]['confirmTrackNum'] = boxiConfirm 
-            }
-            // if specimens exists update, else add following key/values
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('specimens')) {
-              boxIdAndBagsObj[boxIdArray[i]]['specimens'] = boxIdAndBagsObj[boxIdArray[i]]['specimens'] 
-            } 
-            else {
-              boxIdAndBagsObj[boxIdArray[i]] = { '959708259': boxi, confirmTrackNum: boxiConfirm, specimens: boxIdAndBagsObj[boxIdArray[i]] }
-            }  
+            shippingData.push({[conceptIds.shippingTrackingNumber]: trackingId, confirmTrackNum: trackingIdConfirm, boxId});
+            boxIdAndTrackingObj[boxId] = {
+                [conceptIds.shippingTrackingNumber]: trackingId,
+                specimens: boxIdAndBagsObj[boxId],
+              };
         }
 
-        if (isMismatch > - 1) {
+        if (isMismatch) {
             await swal({
```

Comment:
at some point we need to get rid of our reliance on sweet alert (swal)

Reply:
Agreed!